src/cardano.h

```diff
@@ -135,5 +136,18 @@ typedef struct {
 	uint8_t dnsName[DNS_NAME_SIZE_MAX];
 } pool_relay_t;
 
+// ==============================  NATIVE SCRIPTS  ==============================
+
+// depth of n means it can handle up to n-1 levels of nesting
+static const size_t MAX_SCRIPT_DEPTH = 11;
```

Comment:
Do we have a STATIC_ASSERT making sure that levels (i.e. `MAX_SCRIPT_DEPTH`) fits into `uint8_t`? It would fit nicely somewhere into the following piece of code:
```
	// check if we can increase the level without breaking the MAX_SCRIPT_DEPTH constraint
	VALIDATE(ctx->level + 1 < MAX_SCRIPT_DEPTH, ERR_INVALID_DATA);
	ctx->level++;

```
[While this is very unlikely to ever create any problems, we should practice not neglecting such considerations.]

Reply:
Wouldn't it be better to just then use `uint8_t` as the type of the `MAX_SCRIPT_DEPTH`?

Reply:
Yes, true.